### PR TITLE
feat(search): Add inline file search to Svelte app

### PR DESCRIPTION
**Pull request by fkling**
This PR adds the inline file search to the file view.

## Changelog

- Refactored the existing CodeMirror extension to allow UI customization
- Moved utility functions around to avoid loading all of wildcard into the prototype (which lead to build errors)
- Tweaked the new icon component to better align within buttons (I missed that in #62908 because I didn't test it within buttons)
- Noticed that button group styles didn't apply correctly and the reason seemed to be that we were mixing sveltekit and react styles. So I copied the button styles into the sveltekit app.
- Added a switch component, following https://web.dev/articles/building/a-switch-component .

## Test plan

Manually tested the SvelteKit and React versions of the search panel.

### CODE REVIEW

File: client/web-sveltekit/src/lib/wildcard/Button.ts
Comment (fkling) [inline client/web-sveltekit/src/lib/wildcard/Button.ts:1]: This was copied from `@sourcegraph/wildcard` to use the right generated CSS classes.

File: client/web-sveltekit/src/lib/wildcard/Switch.svelte
Comment (fkling) [inline client/web-sveltekit/src/lib/wildcard/Switch.svelte:1]: The new switch component.

File: client/web-sveltekit/src/routes/svelte-overrides.scss [hunk 1]
```diff
@@ -22,6 +22,10 @@ body {
     }
 }
 
+label {
+    margin-bottom: 0;
+}
```
Comment (fkling) [inline client/web-sveltekit/src/routes/svelte-overrides.scss:27]: Don't know why we set a margin by default, but it's rather annoying.

File: client/web/src/repo/blob/codemirror/search.ts
Comment (fkling) [inline client/web/src/repo/blob/codemirror/search.ts:1]: The actual UI was moved into a separate file/component.

File: client/web/src/repo/blob/codemirror/search/SearchPanel.tsx
Comment (fkling) [inline client/web/src/repo/blob/codemirror/search/SearchPanel.tsx:1]: UI that was previously in `search.tsx`.

File: client/web/src/util/dom.ts [hunk 2]
```diff
@@ -99,15 +99,3 @@ export function useBreakpoint(size: keyof typeof breakpoints, debounceMs = 50):
...
-    tagName: K,
-    properties: Partial<HTMLElementTagNameMap[K]> | null = null,
-    ...children: (Node | string)[]
-): HTMLElementTagNameMap[K] {
-    const element = Object.assign(document.createElement(tagName), properties)
-    for (const child of children) {
-        element.append(typeof child === 'string' ? document.createTextNode(child) : child)
-    }
-    return element
-}
```
Comment (fkling) [inline client/web/src/util/dom.ts:113]: This was moved to remove the transitive dependency on wildcard which causes build issues.

**Comment by camdencheek** [comment]
I can't seem to run this locally. Do I need to do anything special to generate the icons?

**Comment by fkling** [comment]
@camdencheek 

> I can't seem to run this locally. Do I need to do anything special to generate the icons?

`pnpm install` and `pnpm dev` should work. What did you try?

**Comment by camdencheek** [comment]
> What did you try?

I tried that 😄

File: client/web-sveltekit/src/lib/wildcard/Button.ts
Comment (fkling) [reply client/web-sveltekit/src/lib/wildcard/Button.ts:1]: The imported `getButtonClassName` function would used the CSS module defined in `@sourcegraph/wildcard`. But the `ButtonGroup.svelte` component actually uses the `Button.module.scss` version in `web-sveltekit`. It seems that the class names are generated differently somehow and bringing in `getButtonClassName` to use the same CSS module as `ButtonGroup.svelte` fixed the problems I was seeing.

Specifically:

| Before | After |
|--------|--------|
|  |  | 

We use the button group for the forward/backward buttons.

File: client/web-sveltekit/src/lib/Icon2.module.scss [hunk 3]
```diff
@@ -7,9 +7,18 @@ svg.icon {
 
     color: var(--icon-fill-color, var(--color, inherit));
     fill: currentColor;
+    stroke-width: var(--stroke-width, 2);
+
+    g,
+    path {
+        // Overwrites Lucide's/unplugin-icons' default stroke-width setting
+        stroke-width: inherit;
+    }
```
[the remark this answers is not in the document]
Comment (fkling) [reply client/web-sveltekit/src/lib/Icon2.module.scss:16]: It seemed quite heavy for smaller icons. Lucide's own API actually allows you make stroke widths absolute or relative to the size: https://lucide.dev/guide/basics/stroke-width.

We'll have to do another pass over icons anyway to tweak sizes and stroke width. Related issue: https://linear.app/sourcegraph/issue/SRCH-464/revamp-icon-sizes

**Comment by fkling** [comment]
Which error/problem do you get? You can also message me in Slack and I'll take a look when I have time.

File: client/web-sveltekit/src/lib/wildcard/Button.ts
Comment (camdencheek) [reply client/web-sveltekit/src/lib/wildcard/Button.ts:1]: Ah, gotcha. Thanks!

File: client/web-sveltekit/src/lib/Icon2.module.scss (hunk 3 above)
[the remark this answers is not in the document]
Comment (camdencheek) [reply client/web-sveltekit/src/lib/Icon2.module.scss:16]: Got it! cool, will save further comments for that issue

**Comment by camdencheek** [comment]
Not sure what changed, but it's fixed now. 

Seems this broke the layout of search results headers:

screenshot-2024-05-29_10-10-40@2x

**Comment by camdencheek** [comment]
Should the keybind be scoped more tightly? Seems a little weird that `Cmd+F` opens the panel when, e.g., I have the top-level search input focused

**Review by camdencheek — APPROVED** [review]

**Comment by camdencheek** [comment]
nit: I don't love the lucide icons for case sensitivity and regex 😕 They seem imbalanced (specifically, the case sensitive icon looks too small), and the regex icon doesn't really look like a regex icon to me.

screenshot-2024-05-29_21-14-55@2x

**Comment by taiyab** [comment]
> (Also, should the text input be taller than the buttons?)

No. They should all be a standard size (probably resolving to 32px height in this case).

**Pull request merged**